cms/static/js/views/utils/view_utils.js

```diff
@@ -173,6 +174,33 @@ define(["jquery", "underscore", "gettext", "js/views/feedback_notification", "js
             return false;
         };
 
+        keywordValidator = (function () {
+            var regexp = /%%+[^%]+%%/g;
+            var keywords = ['%%USER_ID%%', '%%USER_FULLNAME%%', '%%COURSE_DISPLAY_NAME%%', '%%COURSE_END_DATE%%'];
+            var validate = function (string) {
+                var regex_match = string.match(regexp);
+                var found_keywords = regex_match == null ? [] : regex_match;
+                var invalid_keywords = [];
+                var num_found = found_keywords.length;
+                var curr_keyword;
+
+                $.each(found_keywords, function(i, keyword) {
+                    if ($.inArray(keyword, keywords) == -1) {
+                        invalid_keywords.push(keyword);
+                    }
+                });
+
+                return {
+                    'is_valid': invalid_keywords.length == 0,
```

Comment:
Use the strict equality operator `===`.

Or you could write `!invalid_kewords.length`, though Jason will disagree :P (I need to remember to put together a compelling case/response for this convention in dynamic languages)